src/ceph-disk/ceph_disk/main.py

```diff
@@ -3756,6 +3756,25 @@ def main_activate(args):
             )
             osd_data = get_mount_point(cluster, osd_id)
 
+            args.cluster = cluster
+
+            # Check if encrypted block device in journal
+
+            block_dmcrypt_path = os.path.join(osd_data, 'block_dmcrypt')
+
+            if args.dmcrypt and os.path.exists(block_dmcrypt_path):
+                partition = DevicePartition.factory(
+                    path=None,
+                    dev=os.path.join(osd_data, 'block_dmcrypt'),
+                    args=args)
+
+                partition.osd_dm_key = get_dmcrypt_key(
```

Comment:
Setting that should work. However I'm not sure it doesn't break other things such as Ptype lookup etc. So I'd be very careful with this. Alternatively a much more involved but also not quite strange change would be to simply change the way the dmcrypt keys are stored in the monitor. So that they are also stored with the other uuids. Would also allow for distinct encryption keys though there is probably no need for those.